File: 프로그래머스/Lv.2/148653. 마법의 엘리베이터/마법의 엘리베이터.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int storey) {
	int total_magic_stones = 0;

    while (storey != 0) {
        int last_digit = storey % 10;

        if (last_digit > 5) {
            total_magic_stones += (10 - last_digit);
            storey += 10;
        }
        else if (last_digit < 5) {
            total_magic_stones += last_digit;
        }
        else {
            int next_digit = (storey / 10) % 10;
            if (next_digit > 4) {
                storey += 10;
            }
            total_magic_stones += last_digit;
        }

        storey /= 10;
    }

    return total_magic_stones;
}

File: 프로그래머스/Lv.2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int totalMagicStones = 0;

        while (storey != 0) {
            int lastDigit = storey % 10;

            if (lastDigit > 5) {
                totalMagicStones += (10 - lastDigit);
                storey += 10;
            }
            else if (lastDigit < 5) {
                totalMagicStones += lastDigit;
            }
            else {
                int nextDigit = (storey / 10) % 10;
                if (nextDigit > 4) {
                    storey += 10;
                }
                totalMagicStones += lastDigit;
            }

            storey /= 10;
        }

        return totalMagicStones;
    }
}